### PR TITLE
jormungandr: add ecologic tag in scenario new_default

### PR DESCRIPTION
**Pull request by TeXitoi**
To help the understanding of CO2 emission, we propose to add a picto on the journey summary if the journey is ecologic.

An ecologic journey is ecologic if it emits less than 50% of the journey by car.

This first version only tags the journey if we have a direct path by car. In a followup PR, we'll add an approximation based on the flying journey.

### CODE REVIEW

File: source/jormungandr/jormungandr/scenarios/new_default.py [hunk 1]
```diff
@@ -620,6 +642,7 @@ def fill_journeys(self, request_type, api_request, instance):
 
         sort_journeys(pb_resp, instance.journey_order, api_request['clockwise'])
         tag_journeys(pb_resp)
+        journey_filter.delete_journeys((pb_resp,), api_request)
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/scenarios/new_default.py:633]: The next calls to kraken would be create when we haven't yet deleted the filtered journeys, there is a risk that the next call will be based on a "invalid" journey.

**Comment by kinnou02** [comment]
I will need a little context for understand this PR. What is the purpose of this change?

File: source/jormungandr/jormungandr/scenarios/new_default.py [hunk 2]
```diff
@@ -156,11 +156,33 @@ def sort_journeys(resp, journey_order, clockwise):
         resp.journeys.sort(journey_sorter[journey_order](clockwise=clockwise))
 
 
+def _is_car_direct_path(journey):
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/scenarios/new_default.py]: A lot of function of this kind are already in scenarios/helpers.py

**Comment by TeXitoi** [comment]
Description of the PR updated to explain the goal of this PR.

File: source/jormungandr/jormungandr/scenarios/new_default.py (hunk 1 above)
Comment (TeXitoi) [inline source/jormungandr/jormungandr/scenarios/new_default.py:633]: We corrected it.

File: source/jormungandr/jormungandr/scenarios/new_default.py [hunk 3]
```diff
@@ -156,11 +156,33 @@ def sort_journeys(resp, journey_order, clockwise):
...
 def tag_journeys(resp):
     """
     qualify the journeys
     """
-    pass
+    car = next((j for j in resp.journeys if _is_car_direct_path(j)), None)
+    if car is None:
+        return
+    for j in resp.journeys:
+        if not j.HasField('co2_emission'):
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/scenarios/new_default.py]: the ```co2_emission``` of a physical mode is operational in ntfs, what happen if it's not set?

File: source/jormungandr/jormungandr/scenarios/new_default.py [hunk 4]
```diff
@@ -160,7 +160,17 @@ def tag_journeys(resp):
     """
     qualify the journeys
     """
-    pass
+    car = next((j for j in resp.journeys if helpers.is_car_direct_path(j)), None)
+    if car is None:
+        return
+    for j in resp.journeys:
+        if not j.HasField('co2_emission'):
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/scenarios/new_default.py:167]: the `co2_emission` of a physical mode is optional in ntfs, what happen if it's not set?

File: source/jormungandr/jormungandr/scenarios/helpers.py [hunk 5]
```diff
@@ -185,6 +185,17 @@ def is_non_pt_walk(journey):
 def is_non_pt_bike(journey):
     return journey.type == 'non_pt_bike'
 
+def is_car_direct_path(journey):
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/scenarios/helpers.py:188]: This function should be tested

File: source/jormungandr/jormungandr/scenarios/new_default.py (hunk 4 above)
Comment (TeXitoi) [inline source/jormungandr/jormungandr/scenarios/new_default.py:167]: I'll add a test for the field in the car section. If there is no data for car, we'll do not add any ecologic tag. Thus, if you add the CO2 for car, you'll need to add it to all the non 0 other physical mode.

Comment (kinnou02) [inline source/jormungandr/jormungandr/scenarios/new_default.py:167]: seems fair

File: source/jormungandr/jormungandr/scenarios/helpers.py (hunk 5 above)
Comment (TeXitoi) [inline source/jormungandr/jormungandr/scenarios/helpers.py:188]: done

**Comment by TeXitoi** [comment]
@kinnou02 OK?

**Comment by kinnou02** [comment]
Ok for me

**Comment by TeXitoi** [comment]
Then I think you can merge ;-)

**Pull request merged**